coderd/autobuild/lifecycle_executor.go

```diff
@@ -295,6 +313,8 @@ func getNextTransition(
 
 	case isEligibleForDelete(ws, templateSchedule, currentTick):
 		return database.WorkspaceTransitionDelete, database.BuildReasonAutodelete, nil
+	case !shouldBump(ws, latestBuild, latestJob, templateSchedule, currentTick).IsZero():
+		return "", database.BuildReasonBump, nil
```

Comment:
Currently we only bump a workspace build if there are a non-zero number of connections. Will this end up bumping workspace deadlines regardless of connection count?

Reply:
Also, add a test to make sure it doesn't double bump

Reply:
Yes this will bump regardless of connection count. We want to bump if the build lasts until the next autostart time.